pydantic_settings/main.py

```diff
@@ -95,59 +107,14 @@ def _build_values(
             # to an informative error and much better than a confusing error
             return {}
 
-    class Config(BaseConfig):
-        env_prefix: str = ''
-        env_file: Optional[DotenvType] = None
-        env_file_encoding: Optional[str] = None
-        env_nested_delimiter: Optional[str] = None
-        secrets_dir: Optional[StrPath] = None
-        validate_all: bool = True
-        extra: Extra = Extra.forbid
-        arbitrary_types_allowed: bool = True
-        case_sensitive: bool = False
-
-        @classmethod
-        def prepare_field(cls, field: ModelField) -> None:
-            env_names: Union[List[str], AbstractSet[str]]
-            field_info_from_config = cls.get_field_info(field.name)
-
-            env = field_info_from_config.get('env') or field.field_info.extra.get('env')
-            if env is None:
-                if field.has_alias:
-                    warnings.warn(
-                        'aliases are no longer used by BaseSettings to define which environment variables to read. '
-                        'Instead use the "env" field setting. '
-                        'See https://pydantic-docs.helpmanual.io/usage/settings/#environment-variable-names',
-                        FutureWarning,
-                    )
-                env_names = {cls.env_prefix + field.name}
-            elif isinstance(env, str):
-                env_names = {env}
-            elif isinstance(env, (set, frozenset)):
-                env_names = env
-            elif sequence_like(env):
-                env_names = list(env)
-            else:
-                raise TypeError(f'invalid field env: {env!r} ({display_as_type(env)}); should be string, list or set')
-
-            if not cls.case_sensitive:
-                env_names = env_names.__class__(n.lower() for n in env_names)
-            field.field_info.extra['env_names'] = env_names
-
-        @classmethod
-        def customise_sources(
-            cls,
-            settings_cls: Type[BaseSettings],
-            init_settings: PydanticBaseSettingsSource,
-            env_settings: PydanticBaseSettingsSource,
-            dotenv_settings: PydanticBaseSettingsSource,
-            file_secret_settings: PydanticBaseSettingsSource,
-        ) -> Tuple[PydanticBaseSettingsSource, ...]:
-            return init_settings, env_settings, dotenv_settings, file_secret_settings
-
-        @classmethod
-        def parse_env_var(cls, field_name: str, raw_val: str) -> Any:
-            return cls.json_loads(raw_val)
-
-    # populated by the metaclass using the Config class defined above, annotated here to help IDEs only
-    __config__: ClassVar[Type[Config]]
+    model_config = ConfigDict(  # type: ignore[typeddict-item]
```

Comment:
Should we make a subclass of ConfigDict that adds more items? Maybe that doesn't work well.

Probably we'll want to make this work better with the pydantic type hints at some point (annoying that TypedDicts are generally treated as invariant..). But I guess that's outside the scope of a pydantic settings issue.

Reply:
`SettingsConfigDict` added